### PR TITLE
Upgrade To Babel v7

### PR DESCRIPTION
**Pull request by piavgh**
- Add support for public class field syntax: https://babeljs.io/docs/en/babel-plugin-proposal-class-properties
- In .eslintrc.json, I added `no-invalid-this` to suppress error when using the fat arrow syntax

### CODE REVIEW

File: src/client/.eslintrc.json [hunk 1]
```diff
@@ -14,6 +21,12 @@
   "rules": {
     "import/prefer-default-export": 0,
     "no-console": 0,
-    "require-jsdoc" : 0
+    "require-jsdoc": 0,
+    "semi": [
```
Comment (enuchi) [inline src/client/.eslintrc.json]: any reason semi colons are turned off here?

File: src/client/.eslintrc.json [hunk 2]
```diff
@@ -14,6 +21,12 @@
...
+      2,
+      "never"
+    ],
+    "no-invalid-this": 0,
```
Comment (enuchi) [inline src/client/.eslintrc.json]: Is this rule required in order to add support for plugin-proposal-class-properties? Let me know -- if not would prefer to leave out specific linting decisions and let users decide on their preferences.

Comment (piavgh) [reply src/client/.eslintrc.json]: If we don't use this rule, we will get an annoying linting issue like this:

Screenshot 2019-07-11 10:20:47

File: src/client/.eslintrc.json (hunk 1 above)
Comment (piavgh) [reply src/client/.eslintrc.json]: Sorry, it's my personal preference, will remove this semi now

Comment (piavgh) [reply src/client/.eslintrc.json]: I pushed a commit to remove that rule

File: src/client/.eslintrc.json (hunk 2 above)
Comment (enuchi) [reply src/client/.eslintrc.json]: got it, looks good

File: src/client/.eslintrc.json (hunk 1 above)
Comment (enuchi) [reply src/client/.eslintrc.json]: great, thanks!

**Pull request merged**